diff --git a/mod-solo-collections/src/SoloCollectionsCore.cpp b/mod-solo-collections/src/SoloCollectionsCore.cpp
--- a/mod-solo-collections/src/SoloCollectionsCore.cpp
+++ b/mod-solo-collections/src/SoloCollectionsCore.cpp
@@ -449,24 +449,10 @@ public:
     SoloCollectionsCorePlayerScript() : PlayerScript(
         "SoloCollectionsCorePlayerScript", { PLAYERHOOK_ON_LOGIN, PLAYERHOOK_ON_LOGOUT,
             PLAYERHOOK_ON_UPDATE, PLAYERHOOK_ON_LEARN_SPELL, PLAYERHOOK_ON_STORE_NEW_ITEM,
             PLAYERHOOK_ON_CREATE_ITEM, PLAYERHOOK_ON_QUEST_REWARD_ITEM,
             PLAYERHOOK_ON_AFTER_STORE_OR_EQUIP_NEW_ITEM, PLAYERHOOK_ON_EQUIP,
             PLAYERHOOK_ON_LOOT_ITEM, PLAYERHOOK_ON_GROUP_ROLL_REWARD_ITEM,
-            PLAYERHOOK_CAN_PLAYER_USE_PRIVATE_CHAT,
-            PLAYERHOOK_ON_CAN_USE_FLYING_MOUNT_AS_GROUND,
-            PLAYERHOOK_ON_CAN_REPLACE_MOUNT }) { }
-
-    bool OnPlayerCanUseFlyingMountAsGround(Player* player, SpellInfo const* spellInfo,
-        std::uint32_t mapId, std::uint32_t zoneId, std::uint32_t areaId) override
-    {
-        return IsCppBackendOwner() && GetMountCollectionService().CanUseFlyingMountAsGround(
-            player, spellInfo, mapId, zoneId, areaId);
-    }
-
-    bool OnPlayerCanReplaceMount(Player* player, SpellInfo const* spellInfo) override
-    {
-        return IsCppBackendOwner() && GetMountCollectionService().CanReplaceMount(player, spellInfo);
-    }
+            PLAYERHOOK_CAN_PLAYER_USE_PRIVATE_CHAT }) { }
 
     void OnPlayerLogin(Player* player) override
     {
